src/profile/components/errorPage/__tests__/ErrorPage.test.tsx

```diff
@@ -0,0 +1,199 @@
+      return `hideFrontPageButton is '${params.hideFrontPageButton}' but number of buttons is ${frontPageButtonCount}`;
+    }
+
+    return '';
+  };
+
+  it(`renders 
+      - generic title
+      - generic message  
+      - the login button
+      - the frontpage button
+      when location.search is has no affecting params`, async () => {
+    mockUseLocationValue.search = '';
+    result = render(
+      <MemoryRouter>
+        <ErrorPage />
+      </MemoryRouter>
+    );
+    expect(getErrorsFromElementAndParamsMatch({})).toBe('');
+  });
+  it(`renders 
+      - given title
+      - generic message  
+      - the login button
+      - the frontpage button
+      when location.search is has 
+      - a title 
+      and no other affecting params`, async () => {
+    const title = 'test_title';
+    mockUseLocationValue.search = `title=${title}`;
+    result = render(
+      <MemoryRouter>
+        <ErrorPage />
+      </MemoryRouter>
+    );
+    expect(getErrorsFromElementAndParamsMatch({ title })).toBe('');
+  });
+  it(`renders 
+      - given title
+      - given message  
+      - the login button
+      - the frontpage button
+      when location.search has 
+      - a title 
+      - a message 
+      and no other affecting params`, async () => {
+    const title = 'test_title';
+    const message = 'test_message';
+    mockUseLocationValue.search = `title=${title}&message=${message}`;
+    result = render(
+      <MemoryRouter>
+        <ErrorPage />
+      </MemoryRouter>
+    );
+    expect(getErrorsFromElementAndParamsMatch({ title, message })).toBe('');
+  });
+  it(`renders 
+      - given title
+      - given message  
+      - no login button
+      - the frontpage button
+      when location.search has 
+      - a title
+      - a message 
+      - hideLoginButton with a value 
+      and hideFrontPageButton is not set`, async () => {
+    const title = 'test_title';
+    const message = 'test_message';
+    const hideLoginButton = 'anyValue';
+    mockUseLocationValue.search = `title=${title}&message=${message}&hideLoginButton=${hideLoginButton}`;
+    result = render(
+      <MemoryRouter>
+        <ErrorPage />
+      </MemoryRouter>
+    );
+    expect(
+      getErrorsFromElementAndParamsMatch({
+        title,
+        message,
+        hideLoginButton,
+      })
+    ).toBe('');
+  });
+  it(`renders 
+      - given title
+      - given message  
+      - no login button
+      - no frontpage button
+      when location.search has 
+      - a title
+      - a message 
+      - hideLoginButton with a value 
+      - hideFrontPageButton with a value`, async () => {
+    const title = 'test_title';
+    const message = 'test_message';
+    const hideLoginButton = 'anyValue';
+    const hideFrontPageButton = 'anyValue';
+    // eslint-disable-next-line max-len
+    mockUseLocationValue.search = `title=${title}&message=${message}&hideLoginButton=${hideLoginButton}&hideFrontPageButton=${hideFrontPageButton}`;
```

Comment:
Why not split the long line to multiple lines, if linter complains about it?

Reply:
Because the same prop is repeated in all tests,  it is quite long before this and splitting didn't improve readability much.
It is splitted now! :)

Reply:
Just wondering, is there a way to split a string into many lines in JavaScript? Something like in C++:

```c++
const auto myStr = "very long "
    "string";
```

or Python:

```python
my_str = ("very long "
    "string")
```

Reply:
It is possible with template strings:

mockUseLocationValue.search = `title=${title}\
&message=${message}\
&hideLoginButton=${hideLoginButton}\
&hideFrontPageButton=${hideFrontPageButton}`;

The "\" at the end of line, tells js to continue string without a line break. But then cannot have indentation (space/tab) on each line because those would be in the string. And the code would look weird when few lines are not indented at all.

Reply:
Got it. Would string catenation also work?

```javascript
const myStr = "very long " +
    "string";
```

Reply:
Yes, that style works too.